### ISSUE TITLE
Detect when local file is lost before 1x redundancy

### ISSUE THREAD

**Issue by mtlynch** [NebulousLabs/Sia#2488]
Bug Report
==========

Sia should detect if the file it's trying to upload is no longer present on the filesystem.

**To reproduce**:

```
$ head -c 300M </dev/urandom > /tmp/testfile.txt
$ ./siac renter upload /tmp/testfile.txt testfile.txt
Uploaded '/tmp/testfile.txt' as testfile.txt.
$ rm /tmp/testfile.txt
$ ./siac renter -v
...
File size  Available  Progress  Redundancy  Renewing  Sia path
314.57 MB  No           50.00%        0.00  Yes       testfile.txt
```

**Expected**

`./siac renter -v` output indicates that the source file is missing and the upload is stuck.
`renter.log` contains information about the error

**Actual**

`./siac renter -v` does not indicate any problems except that the user will notice no upload progress.
`renter.log` contains a vague error message that does not specify a particular file or say that it's missing:

```
2017/11/25 05:40:51.070595 download.go:501: Not enough workers to finish download: insufficient hosts to recover file
2017/11/25 05:40:51.078194 download.go:501: Not enough workers to finish download: insufficient hosts to recover file
2017/11/25 05:40:51.145722 download.go:501: Not enough workers to finish download: insufficient hosts to recover file
2017/11/25 05:40:51.150447 download.go:501: Not enough workers to finish download: insufficient hosts to recover file
```

This caused a blocking bug in the Sia Minio integration: https://github.com/minio/minio/pull/5233

**Environment**

* Sia version: 47514b49625fe841348d24773ba4c7c54da4fa17
* OS: Ubuntu 16.04 LTS x64

**Comment by MSevey** [comment]
@mtlynch could you confirm if this is still an issue?  if so we can try and fix it in v1.3.4

**Comment by mtlynch** [comment]
Yep, still repros on 1.3.3 RC2.

```text
$ siac renter upload /sia-uploads/testfile.txt testfile.txt && rm uploads/testfile.txt
Uploaded '/sia-uploads/testfile.txt' as testfile.txt.

$ siac renter -v
Renter info:
        Allowance:         650 SC
          Spent Funds:     90.41 SC
            Storage:       178.9 mS
            Upload:        18.69 mS
            Download:      4.504 SC
            Fees:          85.71 SC
          Unspent Funds:   559.6 SC
            Allocated:     134.9 SC
            Unallocated:   424.7 SC

Tracking 5 files:
File size  Available  Uploaded   Progress  Redundancy  Renewing  Sia path
355.86 MB  Yes         1.132 GB   100.00%        2.50  Yes       bbb_sunflower_1080p_60fps_normal.mp4
673.22 MB  Yes         2.139 GB   100.00%        2.50  Yes       bbb_sunflower_2160p_60fps_normal.mp4
150.54 MB  Yes        503.32 MB   100.00%        2.50  Yes       big_buck_bunny_480p_surround-fix.mp4
 104.8 KB  Yes        125.83 MB   100.00%        2.60  Yes       load-test-plan-2018-02-14.pdf
524.29 MB  No               0 B     0.00%        0.00  Yes       testfile.txt
```

I don't see anything about it in `renter.log` except it jumped from 0 to 13 chunks after I ran the upload command:

```text
$ tail sia-data/renter/renter.log
2018/06/04 19:30:06.697448 uploadheap.go:312: Repairing 0 chunks
2018/06/04 19:45:06.725229 uploadheap.go:312: Repairing 0 chunks
2018/06/04 20:00:06.750222 uploadheap.go:312: Repairing 0 chunks
2018/06/04 20:15:06.771982 uploadheap.go:312: Repairing 0 chunks
2018/06/04 20:30:06.826143 uploadheap.go:312: Repairing 0 chunks
2018/06/04 20:45:06.853490 uploadheap.go:312: Repairing 0 chunks
2018/06/04 21:00:06.886397 uploadheap.go:312: Repairing 0 chunks
2018/06/04 21:15:06.904472 uploadheap.go:312: Repairing 0 chunks
2018/06/04 21:30:06.930234 uploadheap.go:312: Repairing 0 chunks
2018/06/04 21:37:30.447047 uploadheap.go:312: Repairing 13 chunks
```

**Comment by MSevey** [comment]
@mtlynch thanks!  we will see if we can't included this fix in v1.3.4
